TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/BotActivity.java

```diff
@@ -201,10 +208,11 @@ public void onServiceDisconnected(ComponentName componentName) {
     };
 
     @Override
-    public void onConnect(String room) {
+    public void onConnect(String room, String serverAddress) {
         debugView.showPermanently(getString(R.string.connected));
         switchableView.setDisplayedChild(1);
 
+        lastServerPreferences.saveLastConnectedServer(serverAddress);
```

Comment:
is this a temporary hack?

otherwise : I would not be saving state inside the View, I would do it inside of the model (looks like it's called a service in your case). 

i.e. inside of `humanTelepresenceService.connectTo(serverAddress)` compose the `humanTelepresenceService` of SharedPreferences/LongtermStorage